### PR TITLE
#175743414 fit(comment thread) added request comment thread

### PR DESCRIPTION
**Pull request by thepatrickniyo**
### What does this PR do?
- implement request comment thread

### Description of Task to be completed? 
- Implement request comment thread
- Adds Dashboard Sidebar
- Adds Dashboard Navigation

### How should this be manually tested?

- Clone the Repo
- Run `git pull origin requestThread` to pull the requestThread branch
- Then run **`npm install`** to install all dependencies
- Run **`npm start`** to start the server
- Run **`npm test`** to run tests
- The browser opens `localhost:3000`
- Navigate to `localhost:3000/requests/thread`

### Expected output
- The request thread is expected to be displayed
- Click on the `menu icon` to trigger the sidebar

## Any background context you want to provide?
- The sidebar navigation items are displayed according to a logged in userType
- user credentials bellow to login in as requester
- Email: `johndoe@gmail.com`  Password: `JohnDoe123`

### What are the relevant Pivotal Tracker stories?
- [#175743414](https://www.pivotaltracker.com/n/projects/2476677/stories/175743414)

#### Screenshots (if appropriate)
- Full page with hidden sidebar
image

- Page view with active sidebar
image

- Responsive view
image
image
image
image

#### Questions:

- N/A

### CODE REVIEW

**Review by ConstantineMaliza — APPROVED** [review]

**Review by SamuelSabasNayo — COMMENTED** [review]
@PatrickNiyogitare28 Renaming a branch with its type (ft, chore, or bug) and pivotal id is a good practice.
Since it paints a picture of the feature before reviewing it.

File: src/__tests__/DashbaordNav.test.js [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+        expect(<DashboardNav />).not.toBeNull()
+    })
+
+    it('elements', () => {
+         <Router>
+             <DashboardNav/>
+         </Router>  
+    })
+})
+
```
Comment (SimonImanigirimpuhwe) [inline src/__tests__/DashbaordNav.test.js]: Kindly revisit **DashboardNav.js** tests.. it seems you tested the existence of document in the DOM but some functionalities are not yet covered.

**Review by SimonImanigirimpuhwe — CHANGES_REQUESTED** [review]

**Comment by thepatrickniyo** [comment]
> @PatrickNiyogitare28 Renaming a branch with its type (ft, chore, or bug) and pivotal id is a good practice.
> Since it paints a picture of the feature before reviewing it.

Responded

**Comment by SamuelSabasNayo** [comment]
**Image is enlarged alot.**
Screen Shot 2021-01-06 at 13 09 29

**Comment by SamuelSabasNayo** [comment]
@PatrickNiyogitare28, you have to rebase your branch to develop.

File: src/__tests__/DashbaordNav.test.js (hunk 1 above)
Comment (thepatrickniyo) [reply src/__tests__/DashbaordNav.test.js]: Hi! kindly review again! now test are raised

**Review by SimonImanigirimpuhwe — APPROVED** [review]
Good job!

**Review by SamuelSabasNayo — APPROVED** [review]

**Comment by Malaba6** [comment]
image
image

**Comment by malaba6** [comment]
This is what's happening on mobile view
image

**Comment by malaba6** [comment]
image

**Comment by malaba6** [comment]
I ran tests and this one failed
image
image

File: src/components/DashboardLayout/DashboardLayout.css [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+/* .contentContainer{
+    display:flex;
+} */
+.childContainer{
+    margin-top: 5em;
+    width:100%;
+   
+}
+
+/* .sidebarContainer{
```
Comment (malaba6) [inline src/components/DashboardLayout/DashboardLayout.css]: Remove commented out code

File: src/components/Request/Table.js [hunk 3]
```diff
@@ -215,15 +216,18 @@ export const RequestTable = ({classes, page, setPage,selected,setSelected,rowsPe
...
 
   const buttons = <div style={{display: 'flex', flexDirection: 'row'}}>
-                  <Fab color="primary" aria-label="edit" style={{width: 35,height: 30, marginRight: 10,}} disabled={selected.length > 1}><EditIcon style={{width: 20, height: 20,}}/></Fab>
-                  <IconButton color="secondary" aria-label="delete" style={{width: 35,height: 35,}} disabled={selected.length > 1}><DeleteIcon style={{width: 25, height: 25,}}/></IconButton>                 
+                  <Fab  color="primary" aria-label="edit" style={{width: 35,height: 30, marginRight: 10,}} disabled={selected.length > 1}><EditIcon style={{width: 20, height: 20,}}/></Fab>
+                  <IconButton   color="secondary" aria-label="delete" style={{width: 35,height: 35,}} disabled={selected.length > 1}><DeleteIcon style={{width: 25, height: 25,}}/></IconButton>                 
+                  <img src={SideForwardIcon} alt="slidebackIcon" onClick={() => history.push('/requests/thread')} style={{width: 35, height: 35, cursor: 'pointer'}} />
+                  {/* <SideForwardIcon onClick={() => history.push('/requests/thread')} style={{width: 35}} /> */}
```
Comment (malaba6) [inline src/components/Request/Table.js]: Here to

**Review by malaba6 — CHANGES_REQUESTED** [review]

**Comment by malaba6** [comment]
You got too many commit messages consider squashing

**Review by ConstantineMaliza — COMMENTED** [review]
for mobile responsiveness, you can look at this angle  check on picture
image

**Comment by thepatrickniyo** [comment]
> for mobile responsiveness, you can look at this angle check on picture
> image

Thank you for feedback @ConstantineMaliza but No mobile device with search screen width! but anyway I have responded to that. You can re-check, thanks

**Pull request merged**